### PR TITLE
BadgeView

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Control used to used to notify users notifications, or status of something.

xct-badgeview

#### Features

* Customize the position of the badge around the badge content.
* Use any kind of content.
* Customize the badge background, border or font size.
* Use animations for the badge text.

#### Using the control

Create an instance for the badge view control, set the **Text** property and the **Content** property.

```
<controls:BadgeView
    BackgroundColor="Red"
    TextColor="White"
    Text="1">
    <Label
        Text="BadgeView"/>
</controls:BadgeView>
```

####  Customization

The font can be customized using the **FontSize**, **FontAttributes**, and **FontFamily** properties.

```
<controls:BadgeView
    BackgroundColor="Red"
    FontAttributes="Bold"
    FontSize="8"
    TextColor="White"
    Text="1">
    <Label
        Text="Font Customization"/>
</controls:BadgeView>
```

The border color of the badge view can be customized using the **BorderColor** property.

```
<controls:BadgeView
    BackgroundColor="Red"
    BorderColor="Orange"
    TextColor="White"
    Text="1">
    <Label
        Text="Border Customization"/>
</controls:BadgeView>
```

####  Customize the badge position

The default **BadgePosition** is TopRight but you can change the position to any corner.

* TopLeft
* TopRight
* BottomLeft
* BottomRight

#### Animations

When increasing or decreasing the Text number, an animation will automatically be applied.

You can customize the animation applied creating a new class and inherit from `IBadgeAnimation`.

```
Task OnAppearing(View badgeView);
Task OnDisappering(View badgeView);
```

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            appearingAnimation.Commit(badgeView, nameof(OnAppearing), length: AnimationLength,
+               finished: (v, t) => tcs.SetResult(true));
+
+            return tcs.Task;
+        }
+
+        public Task OnDisappering(View badgeView)
+        {
+            if (_translationY == null)
+                _translationY = badgeView.TranslationY;
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs:39]: ```_translationY ??= badgeView.TranslationY```

File: XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        protected uint AnimationLength { get; } = 150;
+        protected uint Offset { get; } = 24;
+
+        double? _translationY;
+
+        public Task OnAppearing(View badgeView)
+        {
+            if (_translationY == null)
+                _translationY = badgeView.TranslationY;
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs:16]: ```_translationY ??= badgeView.TranslationY```

File: XamarinCommunityToolkit/Views/BadgeView/BadgeView.shared.cs [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    [ContentProperty(nameof(Content))]
+    public class BadgeView : BaseTemplatedView<Grid>
+    {
+        bool isVisible;
+        bool placementDone;
+
+        public static readonly BindableProperty ContentProperty =
+            BindableProperty.Create(nameof(Content), typeof(View), typeof(BadgeView), null,
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/BadgeView/BadgeView.shared.cs:14]: null can be omitted )

File: XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    public class BadgeAnimation : IBadgeAnimation
+    {
+        protected uint AnimationLength { get; } = 150;
+		protected uint Offset { get; } = 24;
```
Comment (jfversluis) [inline XamarinCommunityToolkit/Views/BadgeView/BadgeAnimation.shared.cs:9]:
```suggestion
        protected uint Offset { get; } = 24;
```

**Review by jfversluis — DISMISSED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**